gui.js

```diff
@@ -323,7 +323,7 @@ onInputKeyDown = function(e) {
         if (!input.classList.contains("multiline")) {
             if (input_history_index > 0) {
                 input_history_index--
-                input.innerHTML = input_history[input_history_index]
+                input.textContent = input_history[input_history_index]
```

Comment:
Hrm, never heard of "textContent".  (Bear in mind, I don't do much web programming...)

Should everything be using `textContent` instead of `innerText`?  A lot of times when I read things that are supposed to answer a question I find I get more confused:

https://stackoverflow.com/questions/35213147/difference-between-textcontent-vs-innertext

It seems we probably need some example of styling of what you're doing as you type in order to help clarify what's-used-where.  Right now the contentEditable is just an abstract "hey, we might want styling someday" feature.  But it's probably better than later to add some dummy examples of styling so we know what's going on.

The history presumably would strip the styling out, and then analyze and re-apply after paste.

Reply:
`innerHTML` is used when you want to render HTML tags in the content, and most of the time this is what you want to do since you're working with an HTML document. However, in this case (since the input div represents a command line) we need the tags to be taken literally and so, to do this, we set `textContent` instead. This causes the browser to do no rendering at all, and it will display the content as plain text. `innerText`, I believe is just a non-standard version of `textContent`, that was really only used in IE for a while.

> The history presumably would strip the styling out, and then analyze and re-apply after paste.

Yea, I think you're right. This would be the best way to do it.